### ISSUE TITLE
Error Creating Swagger doc from json with external references

### ISSUE THREAD

**Issue by DDtMM** [RSuter/NSwag#1287]
NSwag.CodeGeneration.CSharp 11.17.1.

Trying to generate a client from https://developers.adp.com/static/Swagger/worker-demographics-v2-swagger.json.  Schemas are all $refs in the same path.  The thrown exception is below.

I could observe using Fiddler that only the main swagger doc was getting downloaded so I thought the issue might be with `GetUrlAsync` passing the entire url as the documentPath parameter to `GetJsonAsync`.  I made my own version that took the file portion out of the path before making a call.  That didn't make a difference.  

I also tried downgrading NSwag.  When I went down to version 15.4 (after trying 16 and getting the same result) the exception would no longer occur, but the application would block.  I could see on Fiddler the last file downloaded was **workerdemographicmeta-v2-schema.json.**  This was consistent after three trials, and was always the third schema.json file downloaded and the third referenced in the swagger doc.

I find it peculiar that the exception is complaining about the third referenced schema but doesn't seem to download the ones that come before it in the file, even though that's the observed behaviors of previous versions, and that same file is what seemed to cause the lockup in the previous versions.

**Exception**
NotSupportedException: Could not resolve the JSON path '**workerdemographics-v2-schema.json**' because no document path is available.

**Stack Trace**
   at NJsonSchema.JsonReferenceResolver.<ResolveReferenceAsync>d__9.MoveNext()
   at NJsonSchema.JsonReferenceResolver.<ResolveReferenceAsync>d__4.MoveNext()
   at NJsonSchema.JsonSchemaReferenceUtilities.JsonReferenceUpdater.<VisitJsonReferenceAsync>d__5.MoveNext()
   at NJsonSchema.Visitors.JsonReferenceVisitorBase.<VisitAsync>d__6.MoveNext()
   //.... a whole lot of these
   at NJsonSchema.Visitors.JsonReferenceVisitorBase.<VisitAsync>d__6.MoveNext()
   at NJsonSchema.Visitors.JsonReferenceVisitorBase.<VisitAsync>d__4.MoveNext()
   at NJsonSchema.JsonSchemaReferenceUtilities.JsonReferenceUpdater.<VisitAsync>d__4.MoveNext()
   at NJsonSchema.JsonSchemaReferenceUtilities.<UpdateSchemaReferencesAsync>d__0.MoveNext()
   at NJsonSchema.Infrastructure.JsonSchemaSerialization.<FromJsonAsync>d__9`1.MoveNext()
   at NSwag.SwaggerDocument.<FromJsonAsync>d__53.MoveNext()
   at NSwag.SwaggerDocument.<FromUrlAsync>d__55.MoveNext()

**Comment by RSuter** [comment]
Try with v11.17.2

**Comment by DDtMM** [comment]
I tried with 11.17.2.  It locked up like previous versions.  

Interesting thing happened on Fiddler.  When trying to download workerdemographicmeta-v2-schema.json (after downloading 3 files)  the body length was -1 while the application was locked up. The download would finish after I stopped the program.  This was consistent.

I realize I made a mistake in my initial error reporting that on older versions workerdemographicmeta-v2-schema.json would be were the application locked.  Whereas the error in 11.17.1 was workerdemographics-v2-schema.json (no meta in name).  This appears to be corrected.

I should probably open a new ticket, as the error I initially reported appears fixed.
